src/main/java/de/hpi/bpt/argos/parsing/staticData/ArtifactParserImpl.java

```diff
@@ -0,0 +1,44 @@
+package de.hpi.bpt.argos.parsing.staticData;
+
+import de.hpi.bpt.argos.parsing.XMLFileParser;
+import de.hpi.bpt.argos.parsing.XMLSubParserImpl;
+
+/**
+ * {@inheritDoc}
+ * This is an abstract base class.
+ */
+public abstract class ArtifactParserImpl<ArtifactType> extends XMLSubParserImpl implements ArtifactParser<ArtifactType> {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS00119)